src/javascript/JContent/ContentNavigation/NavigationHeader/SwitchersLayout/SiteSwitcher.jsx

```diff
@@ -119,46 +117,38 @@ class SiteSwitcher extends React.Component {
                             className={styles.siteSwitcher}
                             data={sites.map(s => ({label: s.displayName, value: s.path, name: s.name, site: s.site}))}
                             onChange={(e, siteNode) => {
-                                this.onSelectSite(siteNode, currentLang);
-                                this.props.dispatchBatch([
-                                    cmSetPreviewMode(CM_PREVIEW_MODES.EDIT),
-                                    cmSetPreviewState(CM_DRAWER_STATES.HIDE),
-                                    cmSetPreviewSelection(null)
-                                ]);
+                                onSelectSite(siteNode, currentLang);
                                 return true;
                             }}
                         />
                     );
                 }
             }
-            </Query>
-        );
-    }
-}
-
-const mapStateToProps = state => ({
-    siteKey: state.site,
-    currentLang: state.language
-});
-
-const mapDispatchToProps = dispatch => ({
-    selectSite: (siteNode, language) => {
-        dispatch(cmGoto({site: siteNode.name, language}));
-    },
-    dispatchBatch: actions => dispatch(batchActions(actions))
-});
+        </Query>
+    );
+};
 
 SiteSwitcher.propTypes = {
-    selectSite: PropTypes.func.isRequired,
-    siteKey: PropTypes.string.isRequired,
-    currentLang: PropTypes.string.isRequired,
-    notificationContext: PropTypes.object.isRequired,
-    t: PropTypes.func.isRequired,
-    dispatchBatch: PropTypes.func.isRequired
+    // This must return redux action object compatible with dispatch fcn
+    onSelectAction: PropTypes.func,
+    selectorObject: PropTypes.shape({
+        siteKey: PropTypes.string.isRequired,
+        currentLang: PropTypes.string.isRequired
+    }),
+    notificationContext: PropTypes.object.isRequired
+};
+
+SiteSwitcher.defaultProps = {
+    onSelectAction: (siteNode, language) => (batchActions([
+        cmGoto({site: siteNode.name, language}),
+        cmSetPreviewMode(CM_PREVIEW_MODES.EDIT),
+        cmSetPreviewState(CM_DRAWER_STATES.HIDE),
+        cmSetPreviewSelection(null)
+    ])),
+    selectorObject: {
+        siteKey: state => state.site,
+        currentLang: state => state.language
+    }
 };
 
-export default compose(
-    withTranslation(),
-    connect(mapStateToProps, mapDispatchToProps),
-    withNotifications()
-)(SiteSwitcher);
+export default compose(withNotifications())(SiteSwitcher);
```

Comment:
Even if not related, maybe it's worth doing it now - if you upgrade to react-material 3.0.5, you can also use useNotifications()  and get rid of all hoc

Reply:
Sure, I'll do the upgrade.